src/runtime-helpers.js

```diff
@@ -649,25 +706,19 @@ function processPackage (packages, deploymentPackages, deploymentTriggers, param
     }
 
     if (packages[key]['apis']) {
-      Object.keys(packages[key]['apis']).forEach((api) => {
-        if (packages[key]['apis'][api]) {
-          const objectAPI = createApiObject(packages, key, api, ruleAction, arrSequence, namesOnly)
-          objectAPI.name = api
-          apis.push(objectAPI)
-        } else {
-          throw new Error('Arguments to create API not provided')
-        }
+      Object.keys(packages[key]['apis']).forEach((apiName) => {
+        const apiRoutes = createApiRoutes(packages[key], key, apiName, ruleAction, arrSequence, namesOnly)
+        routes = routes.concat(apiRoutes)
```

Comment:
We might be able to overcome the overhead of `concat` in each `forEach` cycle by using `map`?

```js 
routes = Object.keys(packages[key]['apis']).map(apiName => createApiRoutes(packages[key], key, apiName, ruleAction, arrSequence, namesOnly));
```

Reply:
@basecode beautiful!

Reply:
@basecode `flatMap` did the trick

Reply:
@basecode unfortunately `flatMap` is node >=12 only - had to revert.